### PR TITLE
GH-44569: [GLib] Add GArrowDecimal64DataType

### PR DESCRIPTION
**Pull request by hiroyuki-sato**
### Rationale for this change

`arrow::Decimal64Type` data type has been introduced. It is also necessary to support the same data type in GLib.

### What changes are included in this PR?

This PR implements `GArrowDecimal64DataType`.

### Are these changes tested?

YES

### Are there any user-facing changes?

Before this change: `Arrow::DecimalDataType.new(8, 2)` returned `Decimal128DataType`.
After this change: `Arrow::DecimalDataType.new(8, 2)` returns `Decimal64DataType`

* GitHub Issue: #44569

### CODE REVIEW

File: c_glib/arrow-glib/type.h [hunk 1]
```diff
@@ -97,6 +99,7 @@ typedef enum {
   GARROW_TYPE_TIME64,
   GARROW_TYPE_MONTH_INTERVAL,
   GARROW_TYPE_DAY_TIME_INTERVAL,
+  GARROW_TYPE_DECIMAL64,
```
Comment (kou) [inline c_glib/arrow-glib/type.h:102]: Could you use the same order as `arrow::Type::type`? We want to use the same values as `arrow::Type::type`:
https://github.com/apache/arrow/blob/2df7b23af717f505c16fedbef6e4223ac163ac5a/c_glib/arrow-glib/type.h#L74

We can use ` = 44` for now:

https://github.com/apache/arrow/blob/2df7b23af717f505c16fedbef6e4223ac163ac5a/cpp/src/arrow/type_fwd.h#L464-L465

```diff
diff --git a/c_glib/arrow-glib/type.h b/c_glib/arrow-glib/type.h
index 6f33ad64ef..9c2843109e 100644
--- a/c_glib/arrow-glib/type.h
+++ b/c_glib/arrow-glib/type.h
@@ -113,6 +113,8 @@ typedef enum {
   GARROW_TYPE_LARGE_LIST,
   GARROW_TYPE_MONTH_DAY_NANO_INTERVAL,
   GARROW_TYPE_RUN_END_ENCODED,
+  /* TODO: Remove = 44 when we add STRING_VIEW..DECIMAL32. */
+  GARROW_TYPE_DECIMAL64 = 44,
 } GArrowType;
 
 /**
```

Comment (hiroyuki-sato) [reply c_glib/arrow-glib/type.h]: Thanks Fixed. What does `STRING_VIEW` mean by the way?

Comment (kou) [reply c_glib/arrow-glib/type.h]: https://arrow.apache.org/docs/format/Columnar.html#variable-size-binary-view-layout

We should support it in GLib too.

File: c_glib/arrow-glib/type.h [hunk 2]
```diff
@@ -50,6 +50,8 @@ G_BEGIN_DECLS
  * @GARROW_TYPE_TIME64: Exact time encoded with int64, supporting micro- or nanoseconds
  * @GARROW_TYPE_MONTH_INTERVAL: YEAR_MONTH interval in SQL style.
  * @GARROW_TYPE_DAY_TIME_INTERVAL: DAY_TIME interval in SQL style.
+ * @GARROW_TYPE_DECIMAL64: Precision- and scale-based decimal
+ *   type with 64-bit. Storage type depends on the parameters.
```
Comment (kou) [inline c_glib/arrow-glib/type.h:54]: Could you use the same order as the definition? (Could you move this to after the `@GARROW_TYPE_RUN_END_ENCODED`?)

**Comment by mapleFU** [comment]
Just a naive problem, why decimal32 not implemented? Just first not involve it in this pull request?

**Comment by kou** [comment]
Right. We'll work on decimal32 in a separated PR.
https://github.com/apache/arrow/issues/44392

File: c_glib/arrow-glib/type.h (hunk 2 above)
Comment (hiroyuki-sato) [reply c_glib/arrow-glib/type.h]: I changed the order in 8217b31

**Review by kou — APPROVED** [review]
+1

**Pull request merged**